layers/+tools/shell/packages.el

```diff
@@ -309,4 +309,12 @@
         [return] 'vterm--self-insert)
 
       (add-hook 'vterm-mode-hook 'spacemacs/disable-hl-line-mode)
-      (add-hook 'vterm-mode-hook 'spacemacs//inhibit-global-centered-cursor-mode))))
+      (add-hook 'vterm-mode-hook 'spacemacs//inhibit-global-centered-cursor-mode)
+
+      (purpose-set-extension-configuration
+       :vterm
+       (purpose-conf "vterm"
+                     :mode-purposes
+                     '((vterm-mode . terminal))))
+      (add-to-list 'purpose-special-action-sequences
```

Comment:
Changing `purpose-special-action-sequences` like this is an odd choice, what problem does it try to solve? There probably is a better solution. (I don't notice any problem if I remove this `add-to-list` call)

Reply:
Don't remember, will look into it shortly. But now that you point it out it seems odd to me too to have put in `vterm` config this line which seems unrelated to `vterm` 🤔

Reply:
Indeed it worked the same without this line, so I removed it for now. Why did I add this line ? 🤔 

I guess this is why I should have described _why_ I did it in the `set purpose` commit message. Now I just squashed this commit into the larger `add support for vterm` one, it seemed to belong there.

Reply:
Oh, I think it was related to https://github.com/syl20bnr/spacemacs/issues/8171